smart-contracts/contracts/dex-router-osmosis/src/tests/initialize.rs

```diff
@@ -0,0 +1,483 @@
+use std::str::FromStr;
+
+use cosmwasm_schema::cw_serde;
+use cosmwasm_std::{Addr, Attribute, Coin, Decimal, Uint128};
+use osmosis_std::types::cosmos::base::v1beta1::Coin as OsmoCoin;
+use osmosis_std::types::osmosis::concentratedliquidity::v1beta1::{
+    CreateConcentratedLiquidityPoolsProposal, Pool, PoolRecord, PoolsRequest,
+};
+use osmosis_std::types::osmosis::gamm::v1beta1::MsgJoinPool;
+use osmosis_std::types::osmosis::poolmanager::v1beta1::SpotPriceRequest;
+use osmosis_std::types::osmosis::tokenfactory::v1beta1::{MsgCreateDenom, MsgMint};
+
+use osmosis_test_tube::osmosis_std::types::osmosis::gamm::poolmodels::balancer::v1beta1::MsgCreateBalancerPool;
+use osmosis_test_tube::osmosis_std::types::osmosis::gamm::v1beta1::PoolAsset;
+use osmosis_test_tube::Gamm;
+use osmosis_test_tube::{
+    cosmrs::proto::traits::Message,
+    osmosis_std::types::osmosis::concentratedliquidity::{
+        poolmodel::concentrated::v1beta1::MsgCreateConcentratedPool, v1beta1::MsgCreatePosition,
+    },
+    Account, ConcentratedLiquidity, GovWithAppAccess, Module, OsmosisTestApp, PoolManager,
+    SigningAccount, TokenFactory, Wasm,
+};
+
+use crate::msg::{BestPathForPairResponse, ExecuteMsg, InstantiateMsg, QueryMsg};
+
+pub(crate) const ADMIN_BALANCE_AMOUNT: u128 = 3402823669209384634633746074317682114u128;
+pub(crate) const TOKENS_PROVIDED_AMOUNT: &str = "1000000000000";
+pub(crate) const FEE_DENOM: &str = "uosmo";
+pub(crate) const DENOM_BASE: &str = "udydx";
+pub(crate) const DENOM_QUOTE: &str = "uryeth";
+pub(crate) const INTERMEDIATE_BASE: &str = "uinshallah";
+pub(crate) const INTERMEDIATE_QUOTE: &str = "wosmo";
+
+#[cw_serde]
+pub struct PoolWithDenoms {
+    pub pool: u64,
+    pub denom0: String,
+    pub denom1: String,
+}
+
+pub fn default_init() -> (OsmosisTestApp, Addr, Vec<PoolWithDenoms>, SigningAccount) {
+    let app = OsmosisTestApp::new();
+    let tf = TokenFactory::new(&app);
```

Comment:
This can be probably be avoided, we can just init the account with 4 denoms in the balance so we save the creation, mint and assignation of such denoms we need for testing purposes.